builder/openstack/step_run_source_server.go

```diff
@@ -70,6 +70,11 @@ func (s *StepRunSourceServer) Run(_ context.Context, state multistep.StateBag) m
 		Metadata:         s.InstanceMetadata,
 	}
 
+	// check if image filter returned a source image ID and replace
+	if imageID := state.Get("source_image").(string); imageID != "" {
+		serverOpts.ImageRef = imageID
+	}
```

Comment:
The logic should be that `source_image` and `source_image_name` have precedence over `source_image_filter`.

Reply:
"source_image" is set to the StateBag based on a given ID and returned early in [`step_source_image_info.go`](https://github.com/tcarrio/packer/blob/most-recent-image-openstack/builder/openstack/step_source_image_info.go#L33). I'll have the SourceImageName to override the filter set Name after the filter processing is complete.

Reply:
The change requested here has been implemented with new logic in [`step_source_image_info.go`](https://github.com/tcarrio/packer/blob/e1973a4ff99b541594c64434b854ecca4ff3e125/builder/openstack/step_source_image_info.go#L50).